src/config-import.js

```diff
@@ -215,6 +247,164 @@ async function findPresetsFile(entries) {
+  if (!languageCode) {
+    warnings.push(new Error(`invalid translation language ${lang}`))
+    return
+  }
+  for (const [schemaNameRef, languageTranslationsForDocType] of Object.entries(
+    languageTranslations
+  )) {
+    // TODO: push to warnings when removing categories from default config
+    if (!(schemaNameRef === 'fields' || schemaNameRef === 'presets')) {
+      // warnings.push(new Error(`invalid schemaNameRef ${schemaNameRef}`))
+      continue
+    }
+    if (!isRecord(languageTranslationsForDocType)) {
+      warnings.push(new Error('invalid translation for docType object'))
+      continue
+    }
+    yield* translationsForDocType({
+      languageCode,
+      regionCode,
+      schemaNameRef,
+      languageTranslationsForDocType,
+    })
+  }
+}
+
+/** @param {Object} opts
+ * @param {string} opts.languageCode
+ * @param {string | null | undefined} opts.regionCode
+ * @param {ValidDocTypes} opts.schemaNameRef
+ * @param {Record<ValidDocTypes, unknown>} opts.languageTranslationsForDocType
+ */
+function* translationsForDocType({
+  languageCode,
+  regionCode,
+  schemaNameRef,
+  languageTranslationsForDocType,
+}) {
+  for (const [docName, fieldsToTranslate] of Object.entries(
+    languageTranslationsForDocType
+  )) {
+    if (!isRecord(fieldsToTranslate)) {
+      warnings.push(new Error(`invalid translation field`))
+      return
+    }
+    yield* translationForValue({
+      languageCode,
+      regionCode,
+      schemaNameRef,
+      docName,
+      fieldsToTranslate,
+    })
+  }
+}
+
+/**
+ * @param {Object} opts
+ * @param {string} opts.languageCode
+ * @param {string | null | undefined} opts.regionCode
+ * @param {ValidDocTypes} opts.schemaNameRef
+ * @param {string} opts.docName
+ * @param {Record<string,unknown>} opts.fieldsToTranslate
+ */
+function* translationForValue({
+  languageCode,
+  regionCode,
+  schemaNameRef,
+  docName,
+  fieldsToTranslate,
+}) {
+  for (const [fieldRef, message] of Object.entries(fieldsToTranslate)) {
+    let value = {
+      /** @type {'translation'} */
+      schemaName: 'translation',
+      languageCode,
+      regionCode: regionCode || '',
+      schemaNameRef,
+      fieldRef: '',
+      message: '',
+    }
+
+    if (isRecord(message)) {
+      yield* translateMessageObject({ value, message, docName })
+    } else if (typeof message === 'string') {
+      value = { ...value, fieldRef, message }
+      if (!validate(value.schemaName, { ...value, docIdRef: '' })) {
+        warnings.push(new Error(`Invalid translation ${value.message}`))
+        continue
+      }
+
+      yield { name: docName, value }
+    } else {
+      warnings.push(
+        new Error(`Invalid translation message type ${typeof message}`)
+      )
+      continue
+    }
+  }
+}
+/**
+ * @param {Object} opts
+ * @param {any} opts.value
+ * @param {string} opts.docName
+ * @param {Record<string,unknown>} opts.message
+ */
+function* translateMessageObject({ value, message, docName }) {
```

Comment:
Rather than translating arbitrary objects, what do you think about only translating objects of the shapes we expect?